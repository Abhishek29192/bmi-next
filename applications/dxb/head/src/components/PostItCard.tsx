import React from "react";
import { graphql } from "gatsby";
import { PostItCard } from "@bmi-digital/components";
import { Typography } from "@bmi-digital/components";
import { Button } from "@bmi-digital/components";
import { AnchorLink } from "@bmi-digital/components";
import { getClickableActionFromUrl, Data as LinkData } from "./Link";
import { useSiteContext } from "./Site";

export type Props = {
  cardTheme: "pearl" | "blue-900";
  cardSections: Data[];
  Component?: any;
};

export type Data = {
  id: string;
  title: string;
  hasUnderline: boolean;
  description: {
    description: string;
  };
  link: LinkData | null;
  linkType: "button" | "link" | null;
};

const IntegratedPostItCard = ({
  cardTheme,
  cardSections,
  Component = PostItCard
}: Props) => {
  const { countryCode } = useSiteContext();

  return (
    <Component theme={cardTheme}>
      {cardSections.map(
        ({
          id,
          title,
          hasUnderline,
          description,
          link,
          linkType = "button"
        }) => (
          <Component.Section key={id}>
            <Component.Heading
              hasUnderline={hasUnderline}
              hasDarkBackground={cardTheme === "blue-900"}
            >
              {title}
            </Component.Heading>
            <Component.Content>
              <Typography>{description.description}</Typography>
            </Component.Content>
            {link && (
              <Component.Action>
                {linkType === "button" ? (
                  <Button
                    {...(cardTheme === "blue-900"
                      ? { hasDarkBackground: true, variant: "outlined" }
                      : {})}
                    action={getClickableActionFromUrl(
                      link.linkedPage,
                      link.url,
                      countryCode,
                      null,
                      link.label
                    )}
                  >
                    {link.label}
                  </Button>
                ) : (
                  <AnchorLink
                    action={getClickableActionFromUrl(
                      link.linkedPage,
                      link.url,
                      countryCode,
                      null,
                      link.label
                    )}
                  >
                    {link.label}
                  </AnchorLink>
                )}
              </Component.Action>
            )}
          </Component.Section>
        )
      )}
    </Component>
  );
};

export default IntegratedPostItCard;

export const query = graphql`
  # TODO: If this is a Section it should go under the Section type. Have to check because I don't fully understand.
  fragment PostItCardSectionFragment on ContentfulPostItCardSection {
    id
    hasUnderline
    title
    description {
      description
    }
    link {
      ...LinkFragment
    }
  }
`;
