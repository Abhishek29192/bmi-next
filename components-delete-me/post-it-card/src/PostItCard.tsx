import React from "react";
import ColorPair, { Colors } from "@bmi-digital/components/color-pair";
import Typography, {
  Props as TypographyProps
} from "@bmi-digital/components/typography";
import styles from "./PostItCard.module.scss";

type Props = {
  theme?: Colors;
  children: React.ReactNode;
};

const PostItCardHeading = ({ children, ...rest }: TypographyProps) => {
  return (
    <Typography variant="h4" {...rest}>
      {children}
    </Typography>
  );
};

const PostItCardSection = ({ children }: { children: React.ReactNode }) => {
  return <div className={styles["section"]}>{children}</div>;
};

const PostItCardContent = ({ children }: { children: React.ReactNode }) => {
  return <div className={styles["content"]}>{children}</div>;
};

const PostItCardAction = ({ children }: { children: React.ReactNode }) => {
  return <div className={styles["action"]}>{children}</div>;
};

const PostItCard = ({ theme = "pearl", children }: Props) => {
  return (
    <ColorPair theme={theme}>
      <div className={styles["PostItCard"]}>{children}</div>
    </ColorPair>
  );
};

PostItCard.Section = PostItCardSection;
PostItCard.Heading = PostItCardHeading;
PostItCard.Content = PostItCardContent;
PostItCard.Action = PostItCardAction;

export default PostItCard;
