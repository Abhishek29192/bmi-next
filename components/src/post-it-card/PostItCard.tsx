import React from "react";
import ColorPair, { Colors } from "../color-pair/ColorPair";
import Typography, { Props as TypographyProps } from "../typography/Typography";
import styles from "./PostItCard.module.scss";

type Props = {
  theme?: Colors;
  children: React.ReactNode;
};

const PostItCardHeading = ({ children, ...rest }: TypographyProps) => {
  return (
    <Typography variant="h4" {...rest}>
      {children}
    </Typography>
  );
};

const PostItCardSection = ({ children }: { children: React.ReactNode }) => {
  return <div className={styles["section"]}>{children}</div>;
};

const PostItCardContent = ({ children }: { children: React.ReactNode }) => {
  return <div className={styles["content"]}>{children}</div>;
};

const PostItCardAction = ({ children }: { children: React.ReactNode }) => {
  return <div className={styles["action"]}>{children}</div>;
};

const PostItCard = ({ theme = "pearl", children }: Props) => {
  return (
    <ColorPair theme={theme}>
      <div className={styles["PostItCard"]}>{children}</div>
    </ColorPair>
  );
};

PostItCard.Section = PostItCardSection;
PostItCard.Heading = PostItCardHeading;
PostItCard.Content = PostItCardContent;
PostItCard.Action = PostItCardAction;

export default PostItCard;
