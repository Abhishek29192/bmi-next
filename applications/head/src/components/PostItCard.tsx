import React, { useContext } from "react";
import { graphql } from "gatsby";
import PostItCard from "@bmi/post-it-card";
import Typography from "@bmi/typography";
import Button from "@bmi/button";
import AnchorLink from "@bmi/anchor-link";
import { getClickableActionFromUrl, LinkData } from "./Link";
import { SiteContext } from "./Site";

export type Props = {
  cardTheme: "pearl" | "blue-900";
  cardSections: Data[];
  Component?: any;
};

export type Data = {
  id: string;
  title: string;
  hasUnderline: boolean;
  description: {
    description: string;
  };
  link: LinkData | null;
  linkType: "button" | "link" | null;
};

const IntegratedPostItCard = ({
  cardTheme,
  cardSections,
  Component = PostItCard
}: Props) => {
  const { countryCode } = useContext(SiteContext);

  return (
    <Component theme={cardTheme}>
      {cardSections.map(
        ({
          id,
          title,
          hasUnderline,
          description,
          link,
          linkType = "button"
        }) => (
          <Component.Section key={id}>
            <Component.Heading hasUnderline={hasUnderline}>
              {title}
            </Component.Heading>
            <Component.Content>
              <Typography>{description.description}</Typography>
            </Component.Content>
            {link && (
              <Component.Action>
                {linkType === "button" ? (
                  <Button
                    {...(cardTheme === "blue-900"
                      ? { hasDarkBackground: true, variant: "outlined" }
                      : {})}
                    action={getClickableActionFromUrl(
                      link.linkedPage,
                      link.url,
                      countryCode,
                      null,
                      link.label
                    )}
                  >
                    {link.label}
                  </Button>
                ) : (
                  <AnchorLink
                    action={getClickableActionFromUrl(
                      link.linkedPage,
                      link.url,
                      countryCode,
                      null,
                      link.label
                    )}
                  >
                    {link.label}
                  </AnchorLink>
                )}
              </Component.Action>
            )}
          </Component.Section>
        )
      )}
    </Component>
  );
};

export default IntegratedPostItCard;

export const query = graphql`
  # TODO: If this is a Section it should go under the Section type. Have to check because I don't fully understand.
  fragment PostItCardSectionFragment on ContentfulPostItCardSection {
    id
    hasUnderline
    title
    description {
      description
    }
    link {
      ...LinkFragment
    }
  }
`;
