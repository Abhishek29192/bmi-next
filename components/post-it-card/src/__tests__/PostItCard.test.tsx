import React from "react";
import { render } from "@testing-library/react";
import PostItCard from "../";

describe("PostItCard component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <PostItCard>
        <PostItCard.Section>
          <PostItCard.Heading hasUnderline>Heading 1</PostItCard.Heading>
          <PostItCard.Content>Content</PostItCard.Content>
          <PostItCard.Action>Action</PostItCard.Action>
        </PostItCard.Section>
        <PostItCard.Section>
          <PostItCard.Heading hasUnderline>Heading 2</PostItCard.Heading>
          <PostItCard.Content>Content</PostItCard.Content>
          <PostItCard.Action>Action</PostItCard.Action>
        </PostItCard.Section>
      </PostItCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with different color pair", () => {
    const { container } = render(
      <PostItCard theme="blue-900">
        <PostItCard.Section>
          <PostItCard.Heading hasUnderline>Heading 1</PostItCard.Heading>
          <PostItCard.Content>Content</PostItCard.Content>
          <PostItCard.Action>Action</PostItCard.Action>
        </PostItCard.Section>
        <PostItCard.Section>
          <PostItCard.Heading hasUnderline>Heading 2</PostItCard.Heading>
          <PostItCard.Content>Content</PostItCard.Content>
          <PostItCard.Action>Action</PostItCard.Action>
        </PostItCard.Section>
      </PostItCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
