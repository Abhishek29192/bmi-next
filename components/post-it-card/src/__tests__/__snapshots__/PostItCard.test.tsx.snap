// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItCard component renders correctly 1`] = `
<div>
  <div
    class="ColorPair ColorPair--pearl"
  >
    <div
      class="PostItCard"
    >
      <div
        class="section"
      >
        <h4
          class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
        >
          Heading 1
        </h4>
        <div
          class="content"
        >
          Content
        </div>
        <div
          class="action"
        >
          Action
        </div>
      </div>
      <div
        class="section"
      >
        <h4
          class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
        >
          Heading 2
        </h4>
        <div
          class="content"
        >
          Content
        </div>
        <div
          class="action"
        >
          Action
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostItCard component renders with different color pair 1`] = `
<div>
  <div
    class="ColorPair ColorPair--blue-900"
  >
    <div
      class="PostItCard"
    >
      <div
        class="section"
      >
        <h4
          class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
        >
          Heading 1
        </h4>
        <div
          class="content"
        >
          Content
        </div>
        <div
          class="action"
        >
          Action
        </div>
      </div>
      <div
        class="section"
      >
        <h4
          class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
        >
          Heading 2
        </h4>
        <div
          class="content"
        >
          Content
        </div>
        <div
          class="action"
        >
          Action
        </div>
      </div>
    </div>
  </div>
</div>
`;
